src/components/Personalization/decisionsExtractor.js

```diff
@@ -32,22 +32,22 @@ const createDecision = (decision, items) => {
 };
 
```

Comment:
I'd love to do a refactor here. Not important for this PR, but I think it would simplify this code and make it easier to understand. If you think this will work, I'll create a JIRA ticket.

```
const createDecision = (decision, items) => {
  return { id: decision.id, scope: decision.scope, items };
 };

const findMatchingIndex = (decision, item, predicates) => {
  return predicates.reduce((matchingIndex, predicate, currentIndex) => {
    // Check if we already matched one of the predicates
    if (matchingIndex !== predicates.length) {
      return matchingIndex;
    }
    // Check if this predicate matches
    if (predicate({ decision, item })) {
      return currentIndex;
    }
    // This predicate doesn't match so return index for unmatched items
    return predicates.length;
  }, predicates.length);
}

// returns an array of decisions one greater than the number of predicates.
// the predicates are functions that return true or false, and are given {decision, item}
// index 0 holds the decisions that match the first predicate
// index 1 holds the decisions that match the second predicate
// and so forth
// the last index holds the decisions that didn't match any predicate
const extractDecisions = (decisions, ...predicates) => {
  const matchedDecisions = new Array(predicates.length + 1).map(() => []);
  decisions.forEach(decision => {
    const matchedItems = new Array(predicates.length + 1).map(() => []);
    // note: if a decision doesn't have items, it will be dropped.
    const { items = [] } = decision;
    items.forEach(item => {
      matchedItems[findMatchingIndex(decision, item, predicates)].push(item);
    });
    matchedItems.forEach((matchedItemArray, i) => {
      if (isNonEmptyArray(matchedItemArray)) {
        matchedDecisions[i].push(createDecision(decision, matchedItemArray));
      }
    });
  });
  return matchedDecisions;
};

// inside createOnResponseHandler
const [redirectItems, pageWideDomActions, nonPageWideDomActions, otherDecisions] =
  extractDecisions(
    unprocessedDecisions,
    ({ item }) => item.schema === REDIRECT_ITEM,
    ({ item, decision }) => item.schema === DOM_ACTION && decision.scope === PAGE_WIDE_SCOPE,
    ({ item, decision }) => item.schema === DOM_ACTION && decision.scope !== PAGE_WIDE_SCOPE
  );

```

Reply:
yes, I would love to improve that part

Reply:
https://jira.corp.adobe.com/browse/PDCL-5983